File: api/src/Api/ProductCatalog/Requests/PadSeriesCreateRequest.cs
using System;
using System.Collections.Generic;
using DetailingArsenal.Domain.ProductCatalog;

namespace DetailingArsenal.Api.ProductCatalog {
    public class PadSeriesCreateRequest : IDataTransferObject {
        public string Name { get; set; } = null!;
        public Guid BrandId { get; set; }
        public string Texture { get; set; } = null!;
        public string Material { get; set; } = null!;
        public List<string> PolisherTypes { get; set; } = new();
        public List<PadSizeRaw> Sizes { get; set; } = new();
        public List<PadColorCreate> Colors { get; set; } = new();
    }
}

File: app/src/api/product-catalog/data-transfer-objects/requests/pad-series-create-request.ts
import { PadTexture } from '../pad-texture';
import { PadMaterial } from '../pad-material';
import { PolisherType } from '../polisher-type';
import { PadCreateOrUpdate } from './pad-create-or-update';
import { PadSizeCreateOrUpdate } from './pad-size-create-or-update';

export interface PadSeriesCreateRequest {
    name: string;
    brandId: string;
    polisherTypes: PolisherType[];
    sizes: PadSizeCreateOrUpdate[];
    pads: PadCreateOrUpdate[];
}
